spec/src/main/asciidoc/converters.asciidoc

```diff
@@ -60,6 +60,33 @@ The `Config` will use the `Converter` with the highest `Priority` for each targe
 
 A custom `Converter` for a target type of any of the built-in Converters will overwrite the default Converter.
 
+=== Array Converters
+For the built-in converters and custom converters, the corresponding Array converters are provided
```

Comment:
(sorry for the extremely long comment)
Just as a thought, something to mull over.  Yes, we can provide built in support for collections for custom types.  Or we can expect that the converter implementation registers additional converters for those types.  For the built in types, what you're describing makes a lot of sense.  However, when we look the format, do we expect that the entire string is passed in to the converter, or only the segment we identified as being one value?  I wonder if it makes sense for any of the following cases as well:

- The user passes in (via API changes) what the delimiter is
- The converter takes in the raw string value (like every other converter) and returns a `Collection<E>` or `E[]` type, choosing what it knows the delimiter to be (or even trying multiple)
- Now that we're looking at collection types, Do we still only look at the highest config source for this value?  Since right now the config impl is doing the splitting, it may make sense to clarify this
- What happens if in Config 1.0/1.1 a user already registered a converter of type `List` and we introduce this change? Will their converter break?
- `Set<String>` and `List<String>` aren't classes, they're `ParameterizedType`s, as a result they can't be passed into the current API method (`getValue`)

Reply:
What we are trying to do is to expand the current design to cope with more capabilities. With specifying delimitor will introduce API changes. My view is that if a different parameter is needed, a custom written converter can be developed.
>Now that we're looking at collection types, Do we still only look at the highest config source for >this value? Since right now the config impl is doing the splitting, it may make sense to clarify this
The introduction of this new converter should not change the priority on config source. The converter is no different from other builtin converters.

>What happens if in Config 1.0/1.1 a user already registered a converter of type List and we introduce this change? Will their converter break?

No. All builtin converters have the priority of -1. All user defined ones should have a higher priority.

>Set<String> and List<String> aren't classes, they're ParameterizedTypes, as a result they can't >be passed into the current API method (getValue)

This is not supported in programmatic look but only supported in injection model. In the spec, I only mention List and Set will work in the injection model. I will add one more sentence explicitly to say programmatic look up is not support.